src/core/render/emojify.js

```diff
@@ -35,6 +35,8 @@ export function emojify(text, useNativeEmoji) {
       )
       // Mark colons in comments
       .replace(/<!--[\s\S]+?-->/g, m => m.replace(/:/g, '__colon__'))
+      // Mark colons in URIs
+      .replace(/[a-z]{2,}:\/\/[^\s]+/gi, m => m.replace(/:/g, '__colon__'))
```

Comment:
Can change to the following if you would like it to be optional:

```suggestion
      .replace(/([a-z]{2,}:)?\/\/[^\s]+/gi, m => m.replace(/:/g, '__colon__'))
```

Reply:
You're right about ignoring everything inside `<a>` tags. That will miss some shortcodes that I believe people will expect to be processed.

There are a number of scenarios to consider. We should probably start by defining the behavior. For example:

```html
<style>
  @import url("http://domain.com/:smile/file.css");
  @import url('http://domain.com/:smile/file.css');
  @import url(http://domain.com/:smile/file.css);
  @import "http://domain.com/:smile/file.css";
  @import 'http://domain.com/:smile/file.css';

  .myclass::before {
    content: ":smile:";
    background: url("http://domain.com/:smile/");
    background: url('http://domain.com/:smile/');
    background: url(http://domain.com/:smile/);
  }
</style>

<a href="http://domain.com/:smile:/">:smile: http://domain.com/:smile:/ :smile:</a>
<p>:smile: http://domain.com/:smile:/ :smile:</p>

<img src="http://domain.com/:smile:/file.png" alt=":smile:">
<img srcset="http://domain.com/:smile:/file.jpg 320w" src="http://domain.com/:smile:/file.png">

<script src="http://domain.com/:smile:/file.js"></script>
```

Here's how I would assume we would want to final output to look like:

```html
<style>
  @import url("http://domain.com/:smile/file.css");
  @import url('http://domain.com/:smile/file.css');
  @import url(http://domain.com/:smile/file.css);
  @import "http://domain.com/:smile/file.css";
  @import 'http://domain.com/:smile/file.css';

  .myclass::before {
    content: "😄";
    background: url("http://domain.com/:smile/");
    background: url('http://domain.com/:smile/');
    background: url(http://domain.com/:smile/);
  }
</style>

<a href="http://domain.com/:smile:/">😄 http://domain.com/:smile:/ 😄</a>
<p>😄 http://domain.com/:smile:/ 😄</p>

<img src="http://domain.com/:smile:/file.png" title="😄">
<img srcset="http://domain.com/:smile:/file.jpg 320w" src="http://domain.com/:smile:/file.png">

<script src="http://domain.com/:smile:/file.js"></script>
```

I'm probably missing something, but the rules for the above scenarios would be:

1. Ignore emoji shortcodes in `href`, `src`, and `srcset` attributes
1. Ignore emoji shortcodes in `url()` values
1. Ignore emoji shortcodes in text URLs within tags

Thoughts?

Reply:
I think that looks about right. 3 is a useful _catch all_ for URLs anywhere in the content, while 1 and 2 are useful for handling partial URLs (e.g. relative).

For my use case, 3 is sufficient and what this pull request is meant to fix. 1 and 2 are nice to have.

Reply:
I believe we can handle all three scenarios by making a small change to your suggested change above:

```js
.replace(/([a-z]{2,}:)?\/\/[^\s'">)]+/gi, m => m.replace(/:/g, '__colon__'))
```

Tested here: https://regex101.com/r/Paa6KG/1

Reply:
Fixed